crypto/passphrase.c

```diff
@@ -261,10 +261,8 @@ int ossl_pw_get_passphrase(char *pass, size_t pass_size, size_t *pass_len,
         ui_data = data->_.ui_method.ui_method_data;
     }
 
-    if (ui_method == NULL) {
-        ERR_raise(ERR_LIB_CRYPTO, ERR_R_INTERNAL_ERROR);
```

Comment:
Why are you removing this?  It's a perfectly legitimate report

Reply:
Because it is not always an internal error. If you call OSSL_DECODER_from_bio() and not set the passphrase or passphrase callback in any way and try to decode encrypted key, this error will be raised. And the decoder will raise  `ERR_raise(ERR_LIB_PROV, PROV_R_UNABLE_TO_GET_PASSPHRASE);` anyway, if the callback fails.

I could raise a different error, but which one? Perhaps something like `ERR_raise_data(ERR_LIB_CRYPTO, CRYPTO_R_NO_PASSPHRASE_OR_CALLBACK)` ?

Reply:
How about:

``` C
        ERR_raise_data(ERR_LIB_CRYPTO, ERR_R_PASSED_INVALID_ARGUMENT,
                       "No password method specified");
```

Reply:
Thanks for the suggestion. Fixup pushed. @paulidale still OK?